core/templates/dev/head/domain/exploration/InteractionObjectFactory.ts

```diff
@@ -16,122 +16,140 @@
  * @fileoverview Factory for creating new frontend instances of Interaction
  * domain objects.
  */
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
+
+import { AnswerGroupObjectFactory } from
+  'domain/exploration/AnswerGroupObjectFactory';
+import { HintObjectFactory } from 'domain/exploration/HintObjectFactory';
+import { OutcomeObjectFactory } from 'domain/exploration/OutcomeObjectFactory';
+import { SolutionObjectFactory } from
+  'domain/exploration/SolutionObjectFactory';
+
+export class Interaction {
+  answerGroups;
+  confirmedUnclassifiedAnswers;
+  customizationArgs;
+  defaultOutcome;
+  hints;
+  id;
+  solution;
+  constructor(
+      answerGroups, confirmedUnclassifiedAnswers, customizationArgs,
+      defaultOutcome, hints, id, solution) {
+    this.answerGroups = answerGroups;
+    this.confirmedUnclassifiedAnswers = confirmedUnclassifiedAnswers;
+    this.customizationArgs = customizationArgs;
+    this.defaultOutcome = defaultOutcome;
+    this.hints = hints;
+    this.id = id;
+    this.solution = solution;
+  }
 
-require('domain/exploration/AnswerGroupObjectFactory.ts');
-require('domain/exploration/HintObjectFactory.ts');
-require('domain/exploration/OutcomeObjectFactory.ts');
-require('domain/exploration/SolutionObjectFactory.ts');
-
-angular.module('oppia').factory('InteractionObjectFactory', [
-  'AnswerGroupObjectFactory', 'HintObjectFactory', 'OutcomeObjectFactory',
-  'SolutionObjectFactory',
-  function(
-      AnswerGroupObjectFactory, HintObjectFactory, OutcomeObjectFactory,
-      SolutionObjectFactory) {
-    var Interaction = function(
-        answerGroups, confirmedUnclassifiedAnswers, customizationArgs,
-        defaultOutcome, hints, id, solution) {
-      this.answerGroups = answerGroups;
-      this.confirmedUnclassifiedAnswers = confirmedUnclassifiedAnswers;
-      this.customizationArgs = customizationArgs;
-      this.defaultOutcome = defaultOutcome;
-      this.hints = hints;
-      this.id = id;
-      this.solution = solution;
-    };
-
-    Interaction.prototype.setId = function(newValue) {
-      this.id = newValue;
-    };
-
-    Interaction.prototype.setAnswerGroups = function(newValue) {
-      this.answerGroups = newValue;
-    };
-
-    Interaction.prototype.setDefaultOutcome = function(newValue) {
-      this.defaultOutcome = newValue;
-    };
-
-    Interaction.prototype.setCustomizationArgs = function(newValue) {
-      this.customizationArgs = newValue;
-    };
-
-    Interaction.prototype.setSolution = function(newValue) {
-      this.solution = newValue;
-    };
-
-    Interaction.prototype.setHints = function(newValue) {
-      this.hints = newValue;
-    };
+  setId(newValue): void {
```

Comment:
Type of `newValue` and elsewhere in this file?

Reply:
Added.